src/ucp/rndv/proto_rndv.c

```diff
@@ -278,115 +303,134 @@ ucp_proto_rndv_ctrl_init_priv(const ucp_proto_rndv_ctrl_init_params_t *params,
-            memreg_time, ucs_linear_func_make(ctrl_latency, 0.0),
-            params->unpack_time);
-    ucp_proto_perf_range_add_data(ctrl_perf);
-
-    return UCS_OK;
+    proto_config->proto          = ucp_protocols[proto->proto_id];
+    proto_config->priv           = priv;
+    proto_config->ep_cfg_index   = init_params->ep_cfg_index;
+    proto_config->rkey_cfg_index = init_params->rkey_cfg_index;
+    proto_config->select_param   = *select_param;
+    proto_config->init_elem      = proto;
 }
 
-/* Copy performance ranges from the remote protocol add CTRL overheads */
-static ucs_status_t
-ucp_proto_rndv_ctrl_init_parallel_stages(
+/* Probe a rndv_ctrl variant with a given remote protocol */
+static void ucp_proto_rndv_ctrl_variant_probe(
         const ucp_proto_rndv_ctrl_init_params_t *params,
-        ucp_proto_perf_range_t *ctrl_range,
-        ucp_proto_perf_range_t *remote_range,
-        ucp_proto_caps_t *proto_caps, size_t min_length, size_t max_length)
+        ucp_proto_rndv_ctrl_priv_t *rpriv, size_t priv_size,
+        const ucp_proto_select_param_t *remote_select_param,
+        ucp_proto_init_elem_t *remote_proto, const void *remote_priv)
 {
-    const char *variant_name = ucp_proto_perf_node_name(remote_range->node);
-    const char *proto_name   = ucp_proto_id_field(params->super.super.proto_id,
-                                                  name);
-    const ucp_proto_perf_range_t *parallel_stages[2];
-    UCS_STRING_BUFFER_ONSTACK(agg_buf, 256);
-    size_t range_max_length;
+    const char *variant_name = ucp_proto_id_field(remote_proto->proto_id, name);
+    size_t num_elems       = 0;
+    const ucp_proto_perf_t *perf_elems[3];
+    ucp_proto_perf_t *ctrl_perf, *remote_perf;
+    UCS_STRING_BUFFER_ONSTACK(perf_name_buf, 256);
+    size_t cfg_thresh, cfg_priority;
+    ucs_linear_func_t overhead;
+    ucp_proto_perf_t *perf;
     ucs_status_t status;
 
-    ucs_string_buffer_appendf(&agg_buf, "%s"UCP_PROTO_PERF_NODE_NEW_LINE"%s",
-                              proto_name, variant_name);
-
-    do {
-        range_max_length = ucs_min(remote_range->max_length, max_length);
-        if (range_max_length < min_length) {
-            continue;
-        }
+    ucs_string_buffer_appendf(&perf_name_buf,
+                              "%s"UCP_PROTO_PERF_NODE_NEW_LINE"%s",
+                              params->ctrl_msg_name,
+                              ucp_proto_perf_name(remote_proto->perf));
+    ucs_trace("%s proto variant: cfg_thresh=%zu cfg_priority=%u", variant_name,
+              remote_proto->cfg_thresh, remote_proto->cfg_priority);
 
-        ucs_trace("%s: max %zu remote-op %s" UCP_PROTO_PERF_FUNC_TYPES_FMT,
-                  proto_name, remote_range->max_length, variant_name,
-                  UCP_PROTO_PERF_FUNC_TYPES_ARG(remote_range->perf));
+    /* Set send_overheads to the time to send and receive RTS message */
+    overhead = ucs_linear_func_make(params->super.overhead * 2, 0.0);
```

Comment:
maybe the RTS receive overhead should be part of REMOTE_CPU and not the LOCAL_CPU?

Reply:
You mean that we need to add one portion of `params->super.overhead` to LOCAL_CPU and the same portion to LOCAL_CPU  instead of multiplying it here by 2 and adding whole this overhead to LOCAL_CPU?